services/app-web/src/common-code/healthPlanFormDataType/healthPlanFormData.test.ts

```diff
@@ -541,4 +543,166 @@ describe('submission type assertions', () => {
+                        name: 'contract_supporting_that_applies_to_a_rate_also.pdf',
+                        s3URL: 'fakeS3URL',
+                        documentCategories: [
+                            'RATES_RELATED' as const,
+                            'CONTRACT_RELATED' as const,
+                        ],
+                    },
+                    {
+                        name: 'rate_only_supporting_doc.pdf',
+                        s3URL: 'fakeS3URL',
+                        documentCategories: ['RATES_RELATED' as const],
+                    },
+                ],
+                rateDocuments: [
+                    {
+                        name: 'rateDocument.pdf',
+                        s3URL: 'fakeS3URL',
+                        documentCategories: ['RATES' as const],
+                    },
+                ],
+            },
+            testDescription: 'With valid contract and rate related documents',
+            expectedResult: {
+                rateType: undefined,
+                rateDateCertified: undefined,
+                rateDateStart: undefined,
+                rateDateEnd: undefined,
+                rateCapitationType: undefined,
+                rateAmendmentInfo: undefined,
+                rateProgramIDs: [],
+                actuaryContacts: [],
+                documents: [
+                    {
+                        name: 'contract_supporting_that_applies_to_a_rate_also.pdf',
+                        s3URL: 'fakeS3URL',
+                        documentCategories: ['CONTRACT_RELATED' as const],
+                    },
+                ],
+                rateDocuments: [],
+            },
+        },
+        {
+            submission: {
+                ...mockContractAndRateSub,
+                documents: [
+                    {
+                        name: 'rate_only_supporting_doc.pdf',
+                        s3URL: 'fakeS3URL',
+                        documentCategories: ['RATES_RELATED' as const],
+                    },
+                ],
+                rateDocuments: [
+                    {
+                        name: 'rateDocument.pdf',
+                        s3URL: 'fakeS3URL',
+                        documentCategories: ['RATES' as const],
+                    },
+                ],
+            },
+            testDescription: 'With only valid rate related documents',
+            expectedResult: {
+                rateType: undefined,
+                rateDateCertified: undefined,
+                rateDateStart: undefined,
+                rateDateEnd: undefined,
+                rateCapitationType: undefined,
+                rateAmendmentInfo: undefined,
+                rateProgramIDs: [],
+                actuaryContacts: [],
+                documents: [],
+                rateDocuments: [],
+            },
+        },
+    ]
+    test.each(contractOnlyWithValidRateData)(
+        'Submit CONTRACT_ONLY with valid rate data: $testDescription',
```

Comment:
Can we also test that CONTRACT_AND_RATES packages don't lose any data when removeRatesData is called?

Reply:
I added `'does not remove any rate data from CONTRACT_AND_RATES submissionType and submits successfully'` in `submitHealthPlanPackage.test.ts`